pkg/serviceaccounttoken/secret_test.go

```diff
@@ -108,9 +248,14 @@ func TestEnsureSecretForServiceAccount(t *testing.T) {
 				func(action k8stesting.Action) (bool, runtime.Object, error) {
 					ret := action.(k8stesting.CreateAction).GetObject().(*v1.Secret)
 					ret.ObjectMeta.Name = ret.GenerateName + "abcde"
+					ret.ObjectMeta.Annotations = map[string]string{
+						"kubernetes.io/service-account.name": "test",
+					}
 					ret.Data = map[string][]byte{
 						"token": []byte("abcde"),
 					}
+					ret.Type = v1.SecretTypeServiceAccountToken
+
```

Comment:
Nit: I'm not sure why these two items (annotation + type) are added. The template sets these values, so I would think that you wouldn't need to override them here.

In addition, I think this could cause a problem if someone changed the `SecretTemplate` function to remove the annotation or type, since our test mocks will add them regardless. If there's a testing/mocks issue I'm missing let me know.

Reply:
You're right, it's not needed